wbooks/src/App.js

```diff
@@ -0,0 +1,52 @@
+import React, { Component } from 'react';
+import logo from './wbooks_logo.svg';
+import notifications from './notifications.svg';
+import addBook from './add_book.svg';
+import search from './search.svg';
+import bookIcon from './default_book.svg'
+import './App.css';
+import Books from './Books'
+
+class App extends Component {
+
+  constructor() {
+    super();
+    this.state = {value: ''};
+    this.handleFilter = this.handleFilter.bind(this);
+  }
+
+  handleFilter(event) {
+    this.Books.sortByField({filter: event.target.value});
```

Comment:
Instead of changing the entire book list (and loosing the original) think of what would be the simplest way of describing this view state and props.

In this case you can just have the filters as part of the state. Then the books component will receive by props, the books and filter. Then it will filter the list in the render method. (you can extract the filter to another method if you want).

So the books component will look something like: 
```js
class Books extends Component {
  filteredBooks() {
    this.props.books.filter((book) => // filter the books here using this.props.filterType and this.props.filterText)
  }

  render() {
    return (
      <div className="books">
        {this.filteredBooks().map((book) => (
          ... 
        ))}
      </div>      
    );
  }
}
```

Also keep in mind that if you want the search button to work you'll need a few things:
1. You have two different parts in the state: What the user is typing/selecting (type and filter text) and the current filter which is set when the user clicks the search button (type and filter text).
2. You'll need to pass a function to the form `onSubmit` prop. (Remember to use event.preventDefault() to prevent the default behavior of a form).
3. Now that you have the state and how it changes defined, you can filter the books in the component.